features/storage/kvstore/kv_map/kv_map.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "KVStore.h"
+#include "kv_map.h"
+#include "kv_config.h"
+#include <stdlib.h>
+#include "platform/PlatformMutex.h"
+#include "platform/SingletonPtr.h"
+#include "string.h"
+#include "mbed_error.h"
+
+using namespace mbed;
+
+#define  MAX_ATTACHED_KVS 16
+
+typedef struct {
+    char *partition_name;
+    KVStore *kvstore_instance;
+} kv_map_entry_t;
+
+// Attachment table
+kv_map_entry_t kv_map_table[MAX_ATTACHED_KVS];
+int kv_num_attached_kvs;
+
+static int is_initialized = 0;
+static SingletonPtr<PlatformMutex> mutex;
+
+int kv_init()
```

Comment:
maybe better name is kv_map_init?

Reply:
this file has changed to cpp singleton and the name of the method is KVMap::init